repack v5 by webpack  need to install rspack

### Describe the bug

i don't use rspack but when migrate to v5 i got this `Error: Cannot find module '@rspack/core'` and `Error: Cannot find module '@rspack/plugin-react-refresh'`

### System Info

```shell
npx react-native info
error Failed to load configuration of your project.
Error: Cannot find module '@rspack/core'
Require stack:
```

### Re.Pack Version

5.0.0-rc.2

### Reproduction

use webpack

### Steps to reproduce

react-native webpack-start

--- a/.changeset/cold-chefs-confess.md
+++ b/.changeset/cold-chefs-confess.md
@@ -1,0 +1,6 @@
+---
+"@callstack/repack": minor
+"@callstack/repack-init": minor
+---
+
+Include `@rspack/plugin-react-refresh` with Re.Pack instead of requiring user to install it

--- a/apps/tester-app/package.json
+++ b/apps/tester-app/package.json
@@ -32,7 +32,6 @@
     "@react-native/typescript-config": "0.76.1",
     "@rsdoctor/rspack-plugin": "^0.4.5",
     "@rspack/core": "1.0.8",
-    "@rspack/plugin-react-refresh": "1.0.0",
     "@svgr/webpack": "^8.1.0",
     "@swc/helpers": "^0.5.13",
     "@types/jest": "^29.5.12",

--- a/apps/tester-federation-v2/package.json
+++ b/apps/tester-federation-v2/package.json
@@ -29,7 +29,6 @@
     "@react-native-community/cli-platform-android": "15.0.1",
     "@rsdoctor/rspack-plugin": "^0.4.5",
     "@rspack/core": "1.0.8",
-    "@rspack/plugin-react-refresh": "1.0.0",
     "@swc/helpers": "^0.5.13",
     "@types/jest": "^29.5.12",
     "@types/react": "^18.2.51",

--- a/apps/tester-federation/package.json
+++ b/apps/tester-federation/package.json
@@ -36,7 +36,6 @@
     "@react-native-community/cli-platform-android": "15.0.1",
     "@rsdoctor/rspack-plugin": "^0.4.5",
     "@rspack/core": "1.0.8",
-    "@rspack/plugin-react-refresh": "1.0.0",
     "@swc/helpers": "^0.5.13",
     "@types/jest": "^29.5.12",
     "@types/react": "^18.2.51",

--- a/packages/init/src/tasks/addDependencies.ts
+++ b/packages/init/src/tasks/addDependencies.ts
@@ -8,7 +8,6 @@ import logger from '../utils/logger.js';
 
 const rspackDependencies = [
   '@rspack/core@1.0.3', // 1.0.4 breaks sourcemaps
-  '@rspack/plugin-react-refresh',
   'babel-loader', // still needed for codegen
   '@swc/helpers',
   '@callstack/repack',
@@ -18,7 +17,6 @@ const webpackDependencies = [
   'webpack',
   'terser-webpack-plugin',
   'babel-loader',
-  '@rspack/plugin-react-refresh',
   '@callstack/repack',
 ];
 

--- a/packages/repack/package.json
+++ b/packages/repack/package.json
@@ -60,7 +60,6 @@
     "@react-native-community/cli": "*",
     "@react-native-community/cli-types": "*",
     "@rspack/core": ">=1",
-    "@rspack/plugin-react-refresh": ">=1",
     "react-native": ">=0.74",
     "webpack": ">=5.90"
   },
@@ -78,6 +77,7 @@
   "dependencies": {
     "@callstack/repack-dev-server": "workspace:*",
     "@discoveryjs/json-ext": "^0.5.7",
+    "@rspack/plugin-react-refresh": "1.0.0",
     "colorette": "^2.0.20",
     "dedent": "^0.7.0",
     "events": "^3.3.0",

--- a/packages/repack/package.json
+++ b/packages/repack/package.json
@@ -103,7 +103,6 @@
     "@module-federation/enhanced": "0.6.10",
     "@module-federation/sdk": "0.6.10",
     "@rspack/core": "1.0.8",
-    "@rspack/plugin-react-refresh": "1.0.0",
     "@swc/helpers": "0.5.13",
     "@types/dedent": "^0.7.0",
     "@types/jest": "^29.5.12",

